### PR TITLE
added middleware to skip taskflow for project version updates

### PR DESCRIPTION
**Pull request by joefitter**

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const modelTasksRouter = require('./router/model-tasks');
 const decorators = require('./decorators');
 
 const profile = require('./middleware/profile');
+const resolveProjectVersionUpdates = require('./middleware/resolve-project-version-updates');
```
Comment (lennym) [inline lib/api.js]: Java called, it wants its function names back.

File: lib/middleware/resolve-project-version-updates.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const { get, noop } = require('lodash');
+const resolve = require('../hooks/resolve');
+
+module.exports = settings => (req, res, next) => {
+  const model = get(req.body, 'model');
+  const action = get(req.body, 'action');
+  if (model === 'projectVersion' && action === 'update') {
+    return resolve(settings)({ data: req.body, update: noop })
```
Comment (lennym) [inline lib/middleware/resolve-project-version-updates.js]: We'll want to move this outside of the middleware invocation, so the hook is initialised with the settings exactly once.

```js
module.exports = settings => {
  const hook = resolve(settings);
  return (req, res, next) => {
    ...
  }
}
```

File: lib/api.js (hunk 1 above)
Comment (joefitter) [reply lib/api.js]: when you say "Java", do you mean @wheelsandcogs ?

File: lib/middleware/resolve-project-version-updates.js (hunk 2 above)
Comment (joefitter) [reply lib/middleware/resolve-project-version-updates.js]: yes, good point

File: lib/api.js (hunk 1 above)
Comment (lennym) [reply lib/api.js]: :trollface:

Comment (wheelsandcogs) [reply lib/api.js]: Nice function name 👍

**Pull request merged**